src/yamlfix/adapters.py

```diff
@@ -693,3 +705,33 @@ def _restore_jinja_variables(source_code: str) -> str:
             fixed_source_lines.append(line)
 
         return "\n".join(fixed_source_lines)
+
+    def _fix_comment_only_files(self, source_code: str) -> str:
+        """Add a mapping key with an id to the start of the document\
+            to preserve comments."""
+        fixed_source_lines = []
```

Comment:
Now we're running `_fix_comment_only_files` and `_restore_comment_only_files` by default, which slows down the program as it needs to iterate through the source code 2 times. 

I'd only run these functions if the config is set.